### ISSUE TITLE
Should `RUF013` cover type hints in `AnnAssign`?

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#8141]
```python
def f(a: int = None):
       # ^^^ RUF013
    pass


# Good (but should be bad?)
a: int = None


class Foo:
    a: int = None
```

https://play.ruff.rs/96716587-c53e-4175-a1fc-386b0a25966d

**Comment by tmke8** [comment]
I think implicit-optional only ever applied to function definitions, so while the other two assignments clearly violate type rules, they were never accepted by type checkers, so I would say ruff doesn't need a rule for them.

**Comment by harupy** [comment]
Agreed :)
